### PR TITLE
Include Maven custom repo support

### PR DESCRIPTION
**Pull request by keizer619**
```
[[platform.java11.dependency]]
groupId="org.ballerinalang"
artifactId="ballerina-command-distribution"
version="0.8.14"

[[platform.java11.repository]]
id="github-update-tool"
url = "https://maven.pkg.github.com/ballerina-platform/ballerina-update-tool"
username = "{{username}}"
password = "{{password}}"
```

### CODE REVIEW

File: cli/ballerina-cli/src/test/resources/test-resources/validProjectWithCustomMavenRepo/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[package]
+org = "foo"
+name = "winery"
+version = "0.1.0"
+
+[[platform.java11.dependency]]
+groupId="org.ballerinalang"
```
Comment (hevayo) [inline cli/ballerina-cli/src/test/resources/test-resources/validProjectWithCustomMavenRepo/Ballerina.toml:7]: If the same package is in two different repos do we need to flag from which one to resolve ?

File: compiler/ballerina-lang/src/main/resources/ballerina-toml-schema.json [hunk 2]
```diff
@@ -155,7 +155,7 @@
       "properties": {
         "java11": {
           "type": "object",
-          "additionalProperties": false,
+          "additionalProperties": true,
```
Comment (hevayo) [inline compiler/ballerina-lang/src/main/resources/ballerina-toml-schema.json:158]: Why do we need to allow additional properties ?

Comment (keizer619) [reply compiler/ballerina-lang/src/main/resources/ballerina-toml-schema.json]: This is not needed for java11.

File: cli/ballerina-cli/src/test/resources/test-resources/validProjectWithCustomMavenRepo/Ballerina.toml (hunk 1 above)
Comment (keizer619) [reply cli/ballerina-cli/src/test/resources/test-resources/validProjectWithCustomMavenRepo/Ballerina.toml:7]: Usually we do not flag in maven repositories as it stops when resolved from one repository. In that case user has to keep the order of repositories

Comment (hevayo) [reply cli/ballerina-cli/src/test/resources/test-resources/validProjectWithCustomMavenRepo/Ballerina.toml:7]: Ok, shall we improve the test case to have dependency rom maven central as well/

**Review by hevayo — APPROVED** [review]

File: cli/ballerina-cli/src/test/resources/test-resources/validProjectWithCustomMavenRepo/Ballerina.toml (hunk 1 above)
Comment (keizer619) [reply cli/ballerina-cli/src/test/resources/test-resources/validProjectWithCustomMavenRepo/Ballerina.toml:7]: Currently Maven central repo scenarios are covered in Bindgen scenario tests. Maven Central Repo and local .m2 is added as a default repo where user does not need to specify in the toml.
https://github.com/ballerina-platform/ballerina-lang/blob/master/misc/ballerina-bindgen/src/test/resources/mvn-test-resources/balProject/Ballerina.toml

**Review by IMS94 — APPROVED** [review]

**Pull request merged**